### PR TITLE
Add UT to ISCSI volume

### PR DESCRIPTION
**Pull request by jianglingxia**
**What this PR does / why we need it**:
Add UT to ISCSI volume and delete some else branch
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 1]
```diff
@@ -485,9 +485,8 @@ func (rc *reconciler) reconstructVolume(volume podVolume) (*reconstructedVolume,
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		uniqueVolumeName = util.GetUniqueVolumeNameForNonAttachableVolume(volume.podName, plugin, volumeSpec)
 	}
+	uniqueVolumeName = util.GetUniqueVolumeNameForNonAttachableVolume(volume.podName, plugin, volumeSpec)
```
Comment (lanchongyizu) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go]: `else` can't be removed?
If deleted, no matter whether attachablePlugin is `nil` or not, `util.GetUniqueVolumeNameForNonAttachableVolume` be always be called.

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 2]
```diff
@@ -558,15 +557,15 @@ func (rc *reconciler) updateDevicePath(volumesNeedUpdate map[v1.UniqueVolumeName
...
 		glog.Errorf("updateStates in reconciler: could not get node status with error %v", fetchErr)
-	} else {
-		for _, attachedVolume := range node.Status.VolumesAttached {
-			if volume, exists := volumesNeedUpdate[attachedVolume.Name]; exists {
-				volume.devicePath = attachedVolume.DevicePath
-				volumesNeedUpdate[attachedVolume.Name] = volume
-				glog.V(4).Infof("Update devicePath from node status for volume (%q): %q", attachedVolume.Name, volume.devicePath)
-			}
+	}
+	for _, attachedVolume := range node.Status.VolumesAttached {
```
Comment (lanchongyizu) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go]: Same problem to the above.

**Comment by jianglingxia** [comment]
ping @saad-ali  
thanks very much

**Comment by jianglingxia** [comment]
ping @saad-ali 
thanks

**Comment by jianglingxia** [comment]
can label approve? thanks @jingxu97

**Pull request closed without merging**